wireless/bluetooth/bt_atomic.c

```diff
@@ -35,78 +35,57 @@
 
 bt_atomic_t bt_atomic_incr(FAR bt_atomic_t *ptr)
 {
-  irqstate_t flags;
   bt_atomic_t value;
 
-  flags = spin_lock_irqsave(NULL);
-  value = *ptr;
-  *ptr  = value + 1;
-  spin_unlock_irqrestore(NULL, flags);
+  value = (bt_atomic_t)atomic_fetch_add((FAR atomic_char *)ptr, 1);
 
   return value;
 }
 
 bt_atomic_t bt_atomic_decr(FAR bt_atomic_t *ptr)
 {
-  irqstate_t flags;
   bt_atomic_t value;
 
-  flags = spin_lock_irqsave(NULL);
-  value = *ptr;
-  *ptr  = value - 1;
-  spin_unlock_irqrestore(NULL, flags);
+  value = (bt_atomic_t)atomic_fetch_sub((FAR atomic_char *)ptr, 1);
 
   return value;
 }
 
 bt_atomic_t bt_atomic_setbit(FAR bt_atomic_t *ptr, bt_atomic_t bitno)
 {
-  irqstate_t flags;
   bt_atomic_t value;
 
-  flags = spin_lock_irqsave(NULL);
-  value = *ptr;
-  *ptr  = value | (1 << bitno);
-  spin_unlock_irqrestore(NULL, flags);
+  value = (bt_atomic_t)atomic_fetch_or((FAR atomic_char *)ptr, (1 << bitno));
```

Comment:
let's typedef bt_atomic_t to some atomic type to avoid the cast

Reply:
ok